tests/qprogram/test_qprogram.py

```diff
@@ -319,14 +319,56 @@ def test_variable_method_raises_error_if_domain_is_not_scalar_and_type_is_set(se
     def test_operation_with_variable_of_wrong_domain_raises_error(self):
         """Test that any operation when used with a variable of wrong domain raises an error."""
         qp = QProgram()
-        float_scalar = qp.variable(Domain.Scalar, float)
-        int_scalar = qp.variable(Domain.Scalar, int)
-
-        with pytest.raises(ValueError):
-            qp.set_frequency(bus="drive", frequency=float_scalar)
-
-        with pytest.raises(ValueError):
-            qp.set_gain(bus="drive", gain=float_scalar)
-
-        with pytest.raises(ValueError):
-            qp.wait(bus="drive", duration=int_scalar)
+        frequency = qp.variable(Domain.Frequency)
+        phase = qp.variable(Domain.Phase)
+        voltage = qp.variable(Domain.Voltage)
+        time = qp.variable(Domain.Time)
+        scalar = qp.variable(Domain.Scalar, float)
+
+        all_types = {frequency, phase, voltage, time, scalar}
+
+        for var in all_types - {frequency}:
+            with pytest.raises(ValueError):
+                qp.set_frequency(bus="drive", frequency=var)
+
+        for var in all_types - {phase}:
+            with pytest.raises(ValueError):
+                qp.set_phase(bus="drive", phase=var)
+
+        for var in all_types - {voltage}:
```

Comment:
```suggestion
       for var in (all_types - {voltage}):
            with pytest.raises(ValueError):
                qp.set_offset(bus="drive", offset_path0=var, offset_path1=var)
                qp.set_gain(bus="drive", gain=var)
```

Reply:
You could optimise testing with something like this.

Reply:
In this case it is not much gaining and the comment might seem a bit picky, but for future cases just to know this will test all the same.

Reply:
I hate to break it to you, but this wouldn't actually work. 😅 Because `qp.set_offset` would raise the `ValueError`, thus ending execution, and `qp.set_gain` would never be tested. (I had the same thought as you 🥂 )

Reply:
Fair enough, I had that conviction, but as you say, this way we know 100% sure. Approved then.